core/include/thread.h

```diff
@@ -184,9 +184,12 @@ struct _thread {
                                          flags                          */
 #endif
 #if defined(MODULE_CORE_MSG)
-    list_node_t msg_waiters;        /**< threads waiting on message     */
+    list_node_t msg_waiters;        /**< threads waiting for their message
+                                         to be delivered to this thread
+                                         (i.e. all blocking sends)      */
     cib_t msg_queue;                /**< message queue                  */
```

Comment:
Maybe I gravely misunderstood what msg_waiters does, but I don't quite get how this describes its purpose...